From 1aae4267206d1a08d30158bfce99a766550f1afd Mon Sep 17 00:00:00 2001
From: Aishwarya Mallampati <amallampati@google.com>
Date: Mon, 23 Feb 2026 17:40:42 +0000
Subject: [PATCH] Make destAddr nullable

Bug: 459460458
Bug: 459460951
Test: atest
Flag: EXEMPT BUGFIX
Change-Id: I22bb91992ee47f3392a0d495f5cef82fa032a87a
(cherry picked from commit c6e940d710fbd4383f2e999f4d1a2234c0ae11dc)
---
 .../com/android/internal/telephony/TelephonyPermissions.java  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/telephony/common/com/android/internal/telephony/TelephonyPermissions.java b/telephony/common/com/android/internal/telephony/TelephonyPermissions.java
index 44e51e09c293..557613257a5c 100644
--- a/telephony/common/com/android/internal/telephony/TelephonyPermissions.java
+++ b/telephony/common/com/android/internal/telephony/TelephonyPermissions.java
@@ -855,13 +855,13 @@ public final class TelephonyPermissions {
      * and return false if user is not associated with the subscription.
      */
     public static boolean checkSubscriptionAssociatedWithUser(@NonNull Context context, int subId,
-            @NonNull UserHandle callerUserHandle, @NonNull String destAddr) {
+            @NonNull UserHandle callerUserHandle, @Nullable String destAddr) {
         // Skip subscription-user association check for emergency numbers
         TelephonyManager tm = (TelephonyManager) context.getSystemService(
                 Context.TELEPHONY_SERVICE);
         final long token = Binder.clearCallingIdentity();
         try {
-            if (tm != null && tm.isEmergencyNumber(destAddr)) {
+            if (tm != null && destAddr != null && tm.isEmergencyNumber(destAddr)) {
                 Log.d(LOG_TAG, "checkSubscriptionAssociatedWithUser:"
                         + " destAddr is emergency number");
                 return true;
-- 
2.55.0.897.gb25b4bd76c-goog

